app/controllers/api/base_controller/parser/request_adapter.rb

```diff
@@ -59,7 +59,7 @@ def hide?(thing)
         def json_body
           @json_body ||= begin
                            body = @request.body.read if @request.body
-                           body.blank? ? {} : JSON.parse(body)
+                           body.blank? ? {} : Hash(JSON.parse(body))
```

Comment:
Now that we are on Ruby 2.3, we can use Array#to_h which will convert associative arrays into Hashes.  I'm assuming that's why you have this Hash() bit here?

Reply:
I'm actually putting this back for now because it causes other existing tests to fail 😩 . Basically:

* When the request body is empty, `@request.body # => "null"`, which is non-blank, but `JSON.parse("null") # => nil`
* Sometimes we are sending a top-level array (:scream:), which can't be converted implicitly into a hash

I made the test I wrote pending instead, because I don't want to fix that as part of this refactoring: https://github.com/ManageIQ/manageiq-api/pull/143/files#diff-c31ee124141903c07b03421343749667R2